scala-package/core/src/main/scala/ml/dmlc/mxnet/io/NDArrayIter.scala

```diff
@@ -16,46 +21,166 @@ import ml.dmlc.mxnet.{DataIter, NDArray, Shape}
  * the size of data does not match batch_size. Roll over is intended
  * for training and can cause problems if used for prediction.
  */
-class NDArrayIter(data: NDArray, label: NDArray = null,
```

Comment:
In most cases, List should be replaced by Vector (IndexedSeq). see http://docs.scala-lang.org/overviews/collections/performance-characteristics.html